### ISSUE TITLE
Knative: Property `quarkus.knative.revision-auto-scaling.container-concurrency` is ignored

### ISSUE THREAD

**Issue by Sgitario** [quarkusio/quarkus#23832]
### Describe the bug

When using the property `quarkus.knative.revision-auto-scaling.container-concurrency` with a value `5` for example and we generate the Knative manifests, the generated Knative service ignores this value and it always uses `0`:

```yaml
---
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2022-02-21 - 11:09:14 +0000
  labels:
    app.kubernetes.io/version: 1.0.0-SNAPSHOT
    app.kubernetes.io/name: knative-demo
  name: knative-demo
spec:
  template:
    metadata:
      labels:
        app.kubernetes.io/version: 1.0.0-SNAPSHOT
        app.kubernetes.io/name: knative-demo
      name: my-revision
    spec:
      containerConcurrency: 0
      containers:
        - image: jcarvaja/knative-demo:1.0.0-SNAPSHOT
          name: knative-demo
          ports:
            - containerPort: 8080
              name: http1
              protocol: TCP
```

I did check that the decorator `ApplyLocalContainerConcurrencyDecorator` is executed with the right value (`5`), but for some reason, this is overwritten later.

**Comment by Sgitario** [comment]
@iocanel I tried to troubleshot this issue, but I could not find which configurator or decorator is overwritten the container concurrency field, any ideas?

**Comment by iocanel** [comment]
> @iocanel I tried to troubleshot this issue, but I could not find which configurator or decorator is overwritten the container concurrency field, any ideas?

Ok, let me have a look

**Comment by iocanel** [comment]
I've started looking into the issue. WIth a first glance I see that configuration is properlly read and dekorators are correctly applied, stilll the end result is not the expected. 

So, it seems that either the dekorators are not applied or there is a race between dekorators. I'll investigate further.

**Comment by iocanel** [comment]
It  seems that updating dekorate fixing the the `quarkus.knative.revision-auto-scaling.container-concurrency` issue.
- `ImagePullPolicy` is still getting ignored.
- `replicas` does not exist on `KnativeConfig`. @spolti what exactly did you want to use here?

**Comment by spolti** [comment]
Hi @iocanel, the ImagePullPolicy is what I was trying to use to instruct minikube to not try to pull the built image when deploying it.

**Comment by iocanel** [comment]
WIP: https://github.com/ioforks/quarkus/tree/fix-23832-knative-props waiting for dekorate bump.

**Comment by spolti** [comment]
Hi @iocanel do you know if the fixes are already on Quarkus 2.14?
